example/lib/providers/ditto_provider.dart

```diff
@@ -0,0 +1,80 @@
+import 'dart:io';
```

Comment:
We can't import `dart:io` here, since it will break the web build. Again, I think the most straightforward approach (and one we should be encouraging customers to follow) is to upgrade to 4.12.0 and then write:

```dart
final platform = Ditto.currentPlatform;

if (platform case SupportedPlatform.android || SupportedPlatform.ios) {
  // request permissions
}
```